### DISCUSSION TITLE
WAN IP change

### DISCUSSION THREAD

**Discussion by RaveGun** [homeassistant-apps/app-cloudflared#232] [Ideas]
From time to time, because I need to restart my modem, I get a new WAN IP. When this happens, one cannot connect anymore to the home assistant issuance.

Once the add-on is restarted, everything comes back to normal.

Ist there some type of monitoring that can be used to automatically restart the add-on?

Even if it's done from the host/OS side.

Last time it happened I was on vacation. I had to ask somebody to go and check if there is still power. Luckily or not, I switched on a light before it happened. It was still on one week later when I arrived back home. It was ok to check that there was still power, not so great for the light bulb and energy usage.

I have an VPN setup but without a dyndns. And because I stopped the duckdns add-on, I cannot find my new WAN IP if I'm away.

Did I have something wrong configured?

Thank you.

**Comment by rzv-trsc** [comment]
Doesn't really answer your question, but I'd suggest digging into Tailscale as a backup solution to remote-access your HA. 
Just toying around with Cloudflared myself - keeping the existing tailscaled access in parallel as a backup.

LE: just remembered I once came across [this](https://community.home-assistant.io/t/display-public-ip-notify-of-change/74655): might help if you start from that and replace the notification (which is useless if tunnel is down) with a call service for addon restart.

**Reply by RaveGun** [reply]
Nice.
This is also a great option.
Thank you.

**Reply by rzv-trsc** [reply]
Please take some time to share the outcome. I'm considering cloudflared remote access and if this works, even if my provider rarely changes ip's, this automation should prevent problems when away. 😊

**Comment by RaveGun** [comment]
Thanks, I will have a look.

But can I still run the DuckDns add-on just to get the updated IP?
It might take up to an hour to get the updated IP but still better than nothing.
If I know my public IP, then it is easier to connect to the Wireguard form my phone and restart the Cloudflare Add-on.

Thank you.

**Reply by rzv-trsc** [reply]
Most welcome. 
Just edited/updated my post, may have found a solution for you. :-)

**Comment by brenner-tobias** [comment]
@RaveGun Thanks for raising. One of the main reasons we are using the Cloudflare tunnel is because all of our home IPs in Germany are changing every 24h. So this is the main use-case (next to not opening ports in your router) that you are solving with this solution. So really the tunnel should automatically reconnect / just continue to work even if your IP changes. 
If it does not, this means that there is something wrong either on your configuration side or in the add-on itself. Therefor it would be great if you could share your Cloudflare logs after a modem restart for us to understand what is not working properly.

**Reply by RaveGun** [reply]
Ah, understood.

For sure there is something wrong with my configuration. There are too many devices involved in the process that I cannot say I know what they are really doing.

Now that I think about, the modern does not have an IP (it it's configured as bridge mode). It is the router that is connected to the modern that is getting an external IP.

Hmm... If I cannot ping 1.1.1.1, for me it means that the modem is not working. Usually there is a red LED also on in the modern. Otherwise in the Admin Web page, not all the uplink downlink channels are available.

I cannot say that this is the case for sure because, most of the times, when it happened, I was not at home. I just got a call, the internet is not working. So now all adults at home know the drill: go to the basement, pull this one power supply, wait 5 seconds and plug it back in. I should automate this. 

So if I restart the modem, the router maybe gets the same IP or not, but it is clear that if I restart the modem the tunnel is no longer working.

What is the expected downtime once the IP has changed?

Or maybe this works "only" if the IP changes?
If the IP does not change but the connection is interrupted, will the tunnel sense that? Is there an 24/7 socket opened?

I would gladly send the logs, but it's there a description about how and what logs should I get?

Thank you.

**Reply by brenner-tobias** [reply]
So basically it should always work, als long as your HA has a working outbound internet connection. So once the connection drops, the tunnel should try to reestablish the connection for as long as it takes. Your providers IP does not matter in that process and is therefore not affecting the tunnel (so if it changes or not is irrelevant).
Regarding the logs: I suggest you go to the logs after you restarted your modem / router. The setup should be that you have a working internet connection but the tunnel is not working. So you can login locally to HA and get a dump of the Cloudflared Add-On logs.
